var/spack/repos/builtin/packages/scorep/package.py

```diff
@@ -72,6 +74,7 @@ def install(self, spec, prefix):
                 "--with-papi-header=%s" % spec['papi'].prefix.include,
                 "--with-papi-lib=%s" % spec['papi'].prefix.lib,
                 "--enable-shared",
+                "--without-shmem" if '~shmem' in spec else '',
```

Comment:
I would recommend:
```python
"--with{0}-shmem".format('' if '+shmem' in spec else 'out'),
```
Depending on the package, when you don't specify a flag, it will enable it if it can and disable it if it can't. So you end up with non-deterministic packages that may be built one way or another way. It sounds like that isn't the case here (it always defaults to `--with-shmem` even if you don't have shmem). But why not play it on the safe side and explicitly enable/disable it?